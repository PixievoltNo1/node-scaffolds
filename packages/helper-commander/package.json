{
  "name": "@guanghechen/helper-commander",
  "version": "4.1.1",
  "description": "Utils for creating node.js command line program",
  "author": {
    "name": "guanghechen",
    "url": "https://github.com/guanghechen/"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/guanghechen/node-scaffolds/tree/release-4.x.x",
    "directory": "packages/helper-commander"
  },
  "homepage": "https://github.com/guanghechen/node-scaffolds/tree/release-4.x.x/packages/helper-commander#readme",
  "keywords": [
    "commander"
  ],
  "type": "module",
  "module": "./lib/esm/index.mjs",
  "exports": {
    "import": "./lib/esm/index.mjs"
  },
  "types": "lib/types/index.d.ts",
  "source": "src/index.ts",
  "license": "MIT",
  "engines": {
    "node": ">= 16.0.0"
  },
  "files": [
    "lib/",
    "src/",
    "package.json",
    "CHANGELOG.md",
    "LICENSE",
    "README.md"
  ],
  "scripts": {
    "build": "rimraf lib/ && cross-env NODE_ENV=production rollup -c ../../rollup.config.mjs",
    "prepublishOnly": "yarn build",
    "test": "cross-env TS_NODE_FILES=true NODE_OPTIONS=--experimental-vm-modules jest --config ../../jest.config.mjs --rootDir ."
  },
  "dependencies": {
    "@guanghechen/chalk-logger": "^4.1.1",
    "@guanghechen/helper-fs": "^4.1.1",
    "@guanghechen/helper-git": "^4.1.1",
    "@guanghechen/helper-is": "^4.1.1",
    "@guanghechen/helper-option": "^4.1.1",
    "@guanghechen/helper-path": "^4.1.1",
    "@guanghechen/helper-string": "^4.1.1",
    "command-exists": "^1.2.9",
    "commander": "^10.0.0",
    "execa": "^7.0.0",
    "inquirer": "^9.1.4",
    "js-yaml": "^4.1.0"
  },
  "devDependencies": {
    "@types/js-yaml": "^4.0.5",
    "cross-env": "^7.0.3",
    "jest": "^29.4.3",
    "rimraf": "^4.1.2",
    "rollup": "^3.17.0"
  }
}
