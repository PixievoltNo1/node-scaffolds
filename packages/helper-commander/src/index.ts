export * from './command'
export * from './fs'
export * from './git'
export * from './merge'
export * from './option'
export * from './stdin'
export * from './strip-ansi'
export * from './yarn'
