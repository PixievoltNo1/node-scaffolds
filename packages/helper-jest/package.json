{
  "name": "@guanghechen/helper-jest",
  "version": "4.7.3",
  "description": "A collection of utility functions for jest tests",
  "author": {
    "name": "guanghechen",
    "url": "https://github.com/guanghechen/"
  },
  "repository": {
    "type": "git",
    "url": "https://github.com/guanghechen/node-scaffolds/tree/release-4.x.x",
    "directory": "packages/helper-jest"
  },
  "homepage": "https://github.com/guanghechen/node-scaffolds/tree/release-4.x.x/packages/helper-jest#readme",
  "keywords": [],
  "type": "module",
  "main": "./lib/cjs/index.cjs",
  "module": "./lib/esm/index.mjs",
  "exports": {
    "import": "./lib/esm/index.mjs",
    "require": "./lib/cjs/index.cjs"
  },
  "types": "lib/types/index.d.ts",
  "source": "src/index.ts",
  "license": "MIT",
  "engines": {
    "node": ">= 16.0.0"
  },
  "files": [
    "lib/",
    "src/",
    "package.json",
    "CHANGELOG.md",
    "LICENSE",
    "README.md"
  ],
  "scripts": {
    "build": "rimraf lib/ && cross-env NODE_ENV=production rollup -c ../../rollup.config.mjs",
    "prepublishOnly": "yarn build",
    "test": "cross-env TS_NODE_FILES=true NODE_OPTIONS=--experimental-vm-modules jest --config ../../jest.config.mjs --rootDir ."
  },
  "dependencies": {
    "@guanghechen/helper-func": "^4.7.3",
    "@guanghechen/helper-is": "^4.7.3",
    "@jest/globals": "^29.5.0",
    "jest": "^29.5.0"
  },
  "devDependencies": {
    "cross-env": "^7.0.3",
    "rimraf": "^4.4.1",
    "rollup": "^3.20.2"
  }
}
