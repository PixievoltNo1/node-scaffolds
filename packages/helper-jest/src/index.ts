export * from './desensitizer/json'
export * from './desensitizer/types'
export * from './desensitizer/string'
export * from './mock/console'
export * from './mock/logger'
export * from './snapshot/file'
