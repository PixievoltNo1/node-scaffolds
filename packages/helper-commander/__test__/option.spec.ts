import ChalkLogger, { DEBUG } from '@guanghechen/chalk-logger'
import { resolveCommandConfigurationOptions } from '@guanghechen/helper-commander'
import { createLoggerMock } from '@guanghechen/helper-jest'
import { desensitize, locateFixtures } from 'jest.helper'

describe('resolveCommandConfigurationOptions', function () {
  const logger = new ChalkLogger({ level: DEBUG, colorful: false })
  const loggerMock = createLoggerMock({ logger, desensitize })

  beforeEach(() => loggerMock.reset())
  afterAll(() => loggerMock.restore())

  test('basic', function () {
    expect(
      desensitize(
        resolveCommandConfigurationOptions(
          logger,
          '@guanghechen/tool-demo',
          'init',
          locateFixtures('basic2'),
          {
            logLevel: 'info',
            workspace: locateFixtures('basic2'),
            configPath: [locateFixtures('basic2/demo.config.yml')],
            color: true,
            date: false,
            a: [1, 2, 3],
          },
          {
            color: false,
            date: true,
            a: [4, 5],
          },
        ),
      ),
    ).toMatchSnapshot()
    expect(loggerMock.getIndiscriminateAll()).toMatchSnapshot()
  })

  test('with config', function () {
    expect(
      desensitize(
        resolveCommandConfigurationOptions(
          logger,
          '@guanghechen/tool-demo',
          'generate',
          locateFixtures('basic2'),
          {
            logLevel: 'info',
            workspace: locateFixtures('basic2'),
            color: true,
            date: false,
            a: [1, 2, 3],
          },
          {
            color: false,
            date: true,
            a: [4, 5],
          },
        ),
      ),
    ).toMatchSnapshot()
    expect(loggerMock.getIndiscriminateAll()).toMatchSnapshot()
  })

  test('no sub-command', function () {
    expect(
      desensitize(
        resolveCommandConfigurationOptions(
          logger,
          '@guanghechen/tool-demo',
          false,
          locateFixtures('basic2'),
          {
            logLevel: 'warn',
            workspace: locateFixtures('basic2'),
            color: true,
            date: false,
            a: [1, 2, 3],
          },
          {
            color: false,
            date: true,
            a: [4, 5],
          },
        ),
      ),
    ).toMatchSnapshot()
    expect(loggerMock.getIndiscriminateAll()).toMatchSnapshot()
  })
})
