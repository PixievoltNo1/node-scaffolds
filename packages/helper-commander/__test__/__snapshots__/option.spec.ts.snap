// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`resolveCommandConfigurationOptions basic 1`] = `
Object {
  "a": Array [
    1,
    2,
    3,
  ],
  "color": false,
  "configPath": Array [
    "<$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2/demo.config.yml",
  ],
  "cwd": "<$WORKSPACE$>/packages/helper-commander",
  "date": false,
  "logLevel": "debug",
  "parasticConfigEntry": "@guanghechen/tool-demo",
  "parasticConfigPath": "<$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2/package.json",
  "workspace": "<$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2",
}
`;

exports[`resolveCommandConfigurationOptions basic 2`] = `
Array [
  Array [
    "[debug] cwd: <$WORKSPACE$>/packages/helper-commander
",
  ],
  Array [
    "[debug] workspace: <$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2
",
  ],
  Array [
    "[debug] configPath: [
  \\"<$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2/demo.config.yml\\"
]
",
  ],
  Array [
    "[debug] parasticConfigPath: <$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2/package.json
",
  ],
  Array [
    "[debug] parasticConfigEntry: @guanghechen/tool-demo
",
  ],
]
`;

exports[`resolveCommandConfigurationOptions no sub-command 1`] = `
Object {
  "a": Array [
    1,
    2,
    3,
  ],
  "color": true,
  "configPath": Array [],
  "cwd": "<$WORKSPACE$>/packages/helper-commander",
  "date": false,
  "generate": Object {
    "date": true,
    "logLevel": "debug",
  },
  "init": Object {
    "color": true,
    "logLevel": "verbose",
  },
  "logLevel": "warn",
  "parasticConfigEntry": "@guanghechen/tool-demo",
  "parasticConfigPath": "<$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2/package.json",
  "workspace": "<$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2",
}
`;

exports[`resolveCommandConfigurationOptions no sub-command 2`] = `Array []`;

exports[`resolveCommandConfigurationOptions with config 1`] = `
Object {
  "a": Array [
    1,
    2,
    3,
  ],
  "color": true,
  "configPath": Array [],
  "cwd": "<$WORKSPACE$>/packages/helper-commander",
  "date": true,
  "logLevel": "debug",
  "parasticConfigEntry": "@guanghechen/tool-demo",
  "parasticConfigPath": "<$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2/package.json",
  "workspace": "<$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2",
}
`;

exports[`resolveCommandConfigurationOptions with config 2`] = `
Array [
  Array [
    "[debug] cwd: <$WORKSPACE$>/packages/helper-commander
",
  ],
  Array [
    "[debug] workspace: <$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2
",
  ],
  Array [
    "[debug] configPath: []
",
  ],
  Array [
    "[debug] parasticConfigPath: <$WORKSPACE$>/packages/helper-commander/__test__/fixtures/basic2/package.json
",
  ],
  Array [
    "[debug] parasticConfigEntry: @guanghechen/tool-demo
",
  ],
]
`;
